### PR TITLE
Change title and lede on guide pages

### PR DESCRIPTION
**Pull request by stephen-cox**
Subscribes to the PageHeaderDisplayEvent event provided in the feature/localgov-87-title-block branch in localgov_core to modify the title and lede in the PageHeaderBlock block.

Related merge requests:

- Block and event logic: https://github.com/localgovdrupal/localgov_core/pull/25
- Hide page header block for campaign content types: https://github.com/localgovdrupal/localgov_campaigns/pull/27

Discussion of why this block exists and does what is does is here: https://github.com/localgovdrupal/localgov/issues/87

### CODE REVIEW

**Comment by stephen-cox** [comment]
Tests should pass once https://github.com/localgovdrupal/localgov_core/pull/25 is merged.

File: src/EventSubscriber/PageHeaderSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  /**
+   * Set page title and lede.
+   */
+  public function setPageHeader(PageHeaderDisplayEvent $event) {
+    if ($event->getEntity() instanceof Node &&
+      $event->getEntity()->bundle() == 'localgov_guides_page' &&
+      $event->getEntity()->localgov_guides_parent
+    ) {
+      $overview = $event->getEntity()->localgov_guides_parent->entity;
```
Comment (ekes) [inline src/EventSubscriber/PageHeaderSubscriber.php:33]: If it's set, so a entity field list, but an empty one does this return null?
(ie a localgov guide page without the parent set - I assume it's compulsory but)

File: src/EventSubscriber/PageHeaderSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      $event->setTitle($overview->getTitle());
+      if ($overview->get('body')->summary) {
+        $event->setLede([
+          '#type' => 'html_tag',
+          '#tag' => 'p',
+          '#value' => $overview->get('body')->summary,
```
Comment (ekes) [inline src/EventSubscriber/PageHeaderSubscriber.php:39]: My twig is horrid, but I assume this gets the correct filtering automagically?

Comment (stephen-cox) [reply src/EventSubscriber/PageHeaderSubscriber.php:39]: This is a render array processed by the core HtmlTag render (https://git.drupalcode.org/project/drupal/-/blob/8.9.x/core/lib/Drupal/Core/Render/Element/HtmlTag.php) so should be safe :wink:

**Pull request merged**